config/routes.rb

```diff
@@ -30,7 +30,9 @@
   resources :reports, param: :report_id # shallow nesting to avoid id param like :report_report_id
   resources :reports, only: [] do # only: [] to remove duplicate top level routes
     resources :diary_entries, only: [:index, :show]
-    resources :text_components
+    resources :text_components do
+      get 'duplicate', on: :member
```

Comment:
I believe `,on: :member` is unnecessary (default)

Reply:
No, it’s not. Sorry to disappoint you… 😿

Reply:
how about
```ruby
member :text_components do
  get 'duplicate'
end
```
?

Reply:
it's good either way - just leave it like that